go/vt/vtctl/vtctl.go

```diff
@@ -2273,6 +2273,15 @@ func commandVRWorkflow(ctx context.Context, wr *wrangler.Wrangler, subFlags *pfl
 			vrwp.TargetShards = strings.Split(*targetShards, ",")
 			vrwp.SkipSchemaCopy = *skipSchemaCopy
 			vrwp.SourceKeyspace = target
+			// We can't properly handle secondary key deferral when
+			// doing Reshard merges/consolidations (e.g. going from
+			// 2 shards to 1) as you have N vreplication records
+			// and streams running on the same target shard for the
+			// same table.
+			if *deferNonPKeys && len(vrwp.TargetShards) < len(vrwp.SourceShards) {
```

Comment:
This can happen even when there are more source shards than target shards: say, going from `-80, 80-` to `-40, 40-c0, c0-`. If you enable this feature for `TestBasicVreplicationWorkflow` which runs `reshardMerchant2to3SplitMerge` we should see this.

Reply:
Good point! I realized that it's (of course) not Reshard specific either because you can move a table from a sharded keyspace to an unsharded one. That gave me the impetus to add support for these cases in: https://github.com/vitessio/vitess/pull/11700/commits/f9cb3d2092a19e22705041736e980ad036c2fac5